### PR TITLE
#6967: Page Editor home state

### LINKED ISSUE #6967: Page Editor Home State Phase 1-3
- https://www.notion.so/pixiebrix/Page-Editor-Home-State-a2afb27dd7e949199eb253bfce3d18a4?pvs=4

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Closes #6967 
- Adds a Page Editor Home Pane
- Swaps out the PixieBrix Logo for a home icon that shows the Home Pane
- Drops some unused editorSlice code for tracking available mod counts

## Remaining Work

- [x] Fix types - some change made Typescript mad
- [x] Add background image
- [x] Fix/clarify style differences, e.g., fonts
- [x] Fix responsive behavior padding

## Discussion

- The design has some non-standard styles in it, e.g., button padding
- I've reached the limits of my CSS, so likely makes sense for someone else to get the design part across the line

## Demo
https://www.loom.com/share/d1500bdb535d4f5fa1a1b1c7b083c308

## Checklist

- [ ] Add tests
- [ ] New files added to `src/tsconfig.strictNullChecks.json` (if possible)
- [x] Designate a primary reviewer @BLoe

### CODE REVIEW

File: src/pageEditor/panes/HomePane.stories.tsx
Comment (grahamlangford) [inline src/pageEditor/panes/HomePane.stories.tsx:1]: Storybook was throwing errors in this file, so I updated it to match the latest component story format.

https://storybook.js.org/docs/api/csf#upgrading-from-csf-2-to-csf-3

File: src/pageEditor/panes/HomePane.tsx [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                <a href="https://slack.pixiebrix.com/">
+                  Join the Community Slack
+                </a>
+              </li>
+            </ul>
+
+            <ul>
+              <span className={styles.linkSectionHeader}>Helpful Links</span>
+              <li>
+                <button
```
Comment (grahamlangford) [inline src/pageEditor/panes/HomePane.tsx:106]: Converted anchor to button for accessibility reasons.

File: src/tsconfig.strictNullChecks.json [hunk 2]
```diff
@@ -306,19 +306,13 @@
...
     "./pageEditor/panes/RestrictedPane.tsx",
     "./pageEditor/panes/StaleSessionPane.tsx",
-    "./pageEditor/panes/WelcomePane.tsx",
     "./pageEditor/panes/insert/InsertMenuItemPane.tsx",
     "./pageEditor/panes/insert/InsertPanelPane.tsx",
     "./pageEditor/panes/save/savingExtensionSlice.ts",
-    "./pageEditor/sidebar/Logo.tsx",
     "./pageEditor/sidebar/ReloadButton.tsx",
     "./pageEditor/sidebar/SaveButton.tsx",
-    "./pageEditor/sidebar/SidebarCollapsed.tsx",
```
Comment (grahamlangford) [inline src/tsconfig.strictNullChecks.json:321]: `SidebarCollapsed` imports `HomeButton` which imports `editorSlice` which is not strict null checked yet.

File: src/pageEditor/sidebar/HomeButton.module.scss [hunk 3]
```diff
@@ -1,4 +1,4 @@
-/*
+/*!
```
Comment (BLoe) [inline src/pageEditor/sidebar/HomeButton.module.scss:1]: Is this `!` intentional?

**Review by BLoe — APPROVED** [review]
Just a few nit questions, but LGTM

File: src/pageEditor/EditorContent.tsx [hunk 4]
```diff
@@ -130,22 +120,14 @@ const EditorContent: React.FC = () => {
   }
 
   if (isPendingExtensions || isConnectingToContentScript) {
-    // Avoid flashing the panes below while the state is loading. This condition should probably
-    // not be moved below <NoExtensionSelectedPane>, <NoExtensionsPane>, or <WelcomePane>.
+    // Avoid flashing the panes below while the state is loading. This condition could probably
+    // not be moved below <HomePane>
```
[the remark this answers is not in the document]
Comment (grahamlangford) [reply src/pageEditor/EditorContent.tsx:124]: @twschiller

File: src/pageEditor/panes/HomePane.tsx [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+          src={bgIllustration}
+          alt="background illustration"
+          className={styles.bgImage}
+        />
+
+        <Col xs={12} lg="auto">
+          <img src={paintbrush} alt="Page Editor logo" />
+        </Col>
+
+        <Col xs={12} lg={7} xl="auto">
```
[the remark this answers is not in the document]
Comment (grahamlangford) [reply src/pageEditor/panes/HomePane.tsx:48]: Exactly that. `auto` sizes columns based on the "natural width of their content". That caused the links to wrap at the `lg` breakpoint.

File: src/pageEditor/sidebar/HomeButton.module.scss (hunk 3 above)
Comment (grahamlangford) [reply src/pageEditor/sidebar/HomeButton.module.scss:1]: Doubt it. Removing.

File: .git-blame-ignore-revs [hunk 6]
```diff
@@ -0,0 +1 @@
+9c0e8c268d9b800e8b56df7f8d70f0e894d373e2
```
Comment (fregante) [inline .git-blame-ignore-revs:3]: Sneaking in a commit related to Prettier:

PR: https://github.com/pixiebrix/pixiebrix-extension/pull/6792

Docs: https://docs.github.com/en/repositories/working-with-files/using-files/viewing-a-file#ignore-commits-in-the-blame-view

This change improves the future git blames by excluding formatting commits

File: src/pageEditor/EditorContent.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (twschiller) [reply src/pageEditor/EditorContent.tsx:124]: Feel free to revert, I don't feel strongly either way

File: src/pageEditor/EditorContent.tsx [hunk 7]
```diff
@@ -130,22 +120,14 @@ const EditorContent: React.FC = () => {
   }
 
   if (isPendingExtensions || isConnectingToContentScript) {
-    // Avoid flashing the panes below while the state is loading. This condition should probably
-    // not be moved below <NoExtensionSelectedPane>, <NoExtensionsPane>, or <WelcomePane>.
+    // Avoid flashing the panes below while the state is loading. This condition could probably
```
Comment (grahamlangford) [inline src/pageEditor/EditorContent.tsx:123]:
```suggestion
    // Avoid flashing the panes below while the state is loading. This condition should probably
```

**Pull request merged**